### PR TITLE
Add logger and assertions

### PR DESCRIPTION
**Pull request by amitha2210**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/UnmarkAttendanceByStudentCommand.java [hunk 1]
```diff
@@ -59,17 +63,18 @@ public UnmarkAttendanceByStudentCommand(Index targetIndex, Attendance attendance
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        logger.info("Running execute(Model model)");
         List<Person> lastShownList = model.getFilteredPersonList();
 
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            logger.warning(String.format(MESSAGE_LOGGER_FOR_EXCEPTION, UnmarkAttendanceByStudentCommand.class
```
Comment (jinhanjinhan) [inline src/main/java/seedu/address/logic/commands/UnmarkAttendanceByStudentCommand.java:70]: Good use of logger to inform about exceptions thrown, so users will know which part of the code is causing the error

File: src/main/java/seedu/address/logic/commands/UnmarkAttendanceByStudentCommand.java [hunk 2]
```diff
@@ -108,4 +115,27 @@ public String toString() {
...
+     *
+     * @param participationList The list of all participation entries to search through.
+     * @param student The student whose attendance is being marked.
+     * @return The participation record for the specified student.
+     * @throws CommandException if no participation record is found for the specified student.
+     */
+    private Participation getStudentParticipation(List<Participation> participationList,
+                                                  Person student) throws CommandException {
+        return participationList.stream()
+                .filter(participation -> participation.getStudent().equals(student)
```
Comment (jinhanjinhan) [inline src/main/java/seedu/address/logic/commands/UnmarkAttendanceByStudentCommand.java:130]: Good attempt of using SLAP to keep the code shorter and more concised

**Review by jinhanjinhan — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/UnmarkAttendanceByStudentCommandParser.java [hunk 3]
```diff
@@ -15,33 +19,41 @@
...
     public UnmarkAttendanceByStudentCommand parse(String args) throws ParseException {
+        logger.info("Starting to parse arguments");
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_ATTENDANCE, PREFIX_TUTORIAL);
 
         Index index;
 
         try {
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (ParseException e) {
+            logger.warning(String.format(MESSAGE_LOGGER_FOR_EXCEPTION, UnmarkAttendanceByStudentCommandParser.class
```
Comment (Slipperybara) [inline src/main/java/seedu/address/logic/parser/UnmarkAttendanceByStudentCommandParser.java:38]: The exception at parseIndex is already logged, so the logger here will result in duplicate logging, maybe can remove the logger here?

**Review by Slipperybara — APPROVED** [review]
Everything looks good except for the logging at parseIndex

File: src/main/java/seedu/address/logic/parser/UnmarkAttendanceByStudentCommandParser.java (hunk 3 above)
Comment (amitha2210) [reply src/main/java/seedu/address/logic/parser/UnmarkAttendanceByStudentCommandParser.java:38]: Okay! Thanks for pointing it out, will remove

**Pull request merged**